specs/0118-notification-center.md

```diff
@@ -0,0 +1,119 @@
+- Created: 2024-03-12
+- Associated Issue: 
+
+# Summary
+
+Introduce a Notification Center feature to make Pontoon the best tool to send communications to localizers directly, allowing sending of targeted in-app notifications and emails within Pontoon itself.
+
+# Motivation
+
+Currently, we can only send in-app notifications with limited targeting capabilities (selected locales within a single product). For any other communications targeting a subset of users we need to rely on engineers to extract email addresses and send emails to the targeted recipients. This feature would allow administrators of a Pontoon instance managing localization projects to send notifications or emails to targeted sets of recipients with more refined criteria such as locales, projects, roles, recency of activity, and level of activity.
+
+# Feature explanation
+
+## Notification Center
+
+This feature has the ability to compose a message with a subject and body, send a message as either an in-app notification or email, and target a set of recipients based on commonly used criteria.
+
+### Required functionality:
+
+- Title: `Notification Center`
+- Header: `Message type`
+  - Selects the type of message to be sent
+  - Checkboxes: `Email`, `Notification` (multiselect, default: both boxes empty)
+  - Validation: Error if no checkbox selected - `You must select at least one message type.`
+- Header: `Message editor`
+  - Creates a subject and body for emails. Subject line will be required for in-app messages; in app-messages have the subject line combined with the body message as the first line of the message separated by a newline.
+  - Field: `Subject` (text input)
+  - Field: `Body` (text input with formatting: markdown or html)
+  - Button: `Preview` - shows a rendered version of the text in `Body` for checking formatting, similar to how the preview for GitHub comments work
+  - Validation: 
+    - Error if `Subject` field empty: `Your message must include a subject.`
+    - Error if `Body` field empty: `Your message must include a body.`
+- Header: `Recipients`
+  - Select who will receive a message based on a combination of the below filters.
+  - Checkbox: `Filter by User Role` (default empty) - enables User Role as a filter.
+    - Checkboxes: `Manager`, `Translator`, `Contributor` (multiselect, default: all boxes empty)
+    - Validation:
+      - Error if `Filter by User Role` enabled but no roles selected: `You must select at least one user role.`
+  - Checkbox: `Filter by Locale` (default empty) - enables Locale as a filter.
+    - Use existing 2-panel widget already used for selecting locales in Pontoon
+    - Choose locales by moving them from `Available` to `Chosen`(all locales available)
+    - Include a `Move All` option
+    - Validation:
+      - Error if `Filter by Locale` enabled but no locales selected: `You must select at least one locale.`
+  - Checkbox: `Projects`
+    - Use existing 2-panel widget already used for selecting locales in Pontoon
```

Comment:
```suggestion
    - Use existing 2-panel widget like the one already used for selecting locales in Pontoon
```

Maybe this is less confusing, since this is about projects.